Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -2586,19 +2604,35 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     page.process().send(Messages::WebExtensionContextProxy::AddTabPageIdentifier(page.webPageID(), tab.identifier(), windowIdentifier), identifier());
 }
 
+void WebExtensionContext::enumerateExtensionPages(Function<bool(WebPageProxy&)>&& action)
+{
+    if (!isLoaded())
+        return;
+
+    for (auto& page : extensionController()->allPages()) {
+        auto* webView = page.cocoaView().get();
+        if (isURLForThisExtension(webView._requiredWebExtensionBaseURL)) {
+            bool shouldContinue = action(page);
+            if (!shouldContinue)
+                return;
+        }
+    }
+}
+
 WKWebView *WebExtensionContext::relatedWebView()
 {
     ASSERT(isLoaded());
 
     if (m_backgroundWebView)
         return m_backgroundWebView.get();
 
-    for (auto& page : extensionController()->allPages()) {
-        if (isURLForThisExtension(page.configuration().requiredWebExtensionBaseURL()))
-            return page.cocoaView().get();
-    }
+    WKWebView *extensionWebView;
+    enumerateExtensionPages([&](auto& page) {
+        extensionWebView = page.cocoaView().get();
+        return true;
```

Comment:
And for `relatedWebView`, this one seems like it should be false, since we only need one.